lib/post_request_task/task.py

```diff
@@ -69,12 +99,13 @@ def apply_async(self, *args, **kwargs):
 
 
 # Hook the signal handlers up.
-# Send the tasks to celery when the request is finished.
-request_finished.connect(_send_tasks,
+# Start queuing the tasks only if we're inside a request-response cycle thread.
+request_started.connect(_start_queuing_tasks,
+                        dispatch_uid='request_started_tasks')
+
+# Send the tasks to celery and stop queuing when the request is finished.
+request_finished.connect(_send_tasks_and_stop_queuing,
                          dispatch_uid='request_finished_tasks')
-# Also send the tasks when a task is finished (outside the request-response
-# cycle, when a task calls another task).
-task_postrun.connect(_send_tasks, dispatch_uid='tasks_finished_tasks')
 
 # And make sure to discard the task queue when we have an exception in the
 # request-response cycle.
```

Comment:
All the above changes make so much more sense. I like that we enable/disable this task queueing stuff with the request signals starting and stopping. Nicely done.